File: semantic-streams-core/src/main/java/es/upm/fi/dia/oeg/common/ParameterUtils.java
package es.upm.fi.dia.oeg.common;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStream;
import java.net.URISyntaxException;
import java.net.URL;
import java.util.Properties;



public class ParameterUtils 
{
	public static String COMMENT_CHAR = "#";
	public static Properties load(InputStream fis) throws IOException
	{
        Properties props = new Properties();
        props.load(fis);    
        fis.close();
        return props;
	}
    public static Properties load(File propsFile) throws IOException 
    {
        FileInputStream fis = new FileInputStream(propsFile);     
        return load(fis);
    }
    
    public static String loadAsString(URL url) throws IOException, URISyntaxException
    {
    	FileReader fr = new FileReader(new File(url.toURI()));
		BufferedReader br = new BufferedReader(fr);
		String query = "";
		String buffer = "";		
		while((buffer= br.readLine()) != null)
		{
			if (!buffer.startsWith(COMMENT_CHAR))
			query = query + (buffer!=null?buffer+" \n ":" ");
		}		
		return query;	
    }
    
    public static String loadQuery(String path)
	{
		try {
			return ParameterUtils.loadAsString(ParameterUtils.class.getClassLoader().getResource(path));
		} catch (IOException e) {
			e.printStackTrace();
		} catch (URISyntaxException e) {
			e.printStackTrace();
		}
		return null;			
	}
	
   
}

File: sparql-stream/src/main/scala/es/upm/fi/oeg/morph/common/ParameterUtils.scala
package es.upm.fi.oeg.morph.common

import java.util.Properties
import java.io.InputStream
import scala.io.Source
import java.io.File
import java.io.FileInputStream
import java.net.URL

object ParameterUtils {
  private val COMMENT_CHAR = "#"
  def load(fis:InputStream):Properties ={
    val props = new Properties
    props.load(fis)
    fis.close
    return props
  }
  def load(propsFile:File):Properties= 
    load(new FileInputStream(propsFile))     
  
    
  def loadAsString(url:URL):String=  {
    val fr = Source.fromFile(url.toURI())
    fr.getLines.filterNot(_.startsWith(COMMENT_CHAR)).mkString("\n")
  }
    
  def loadQuery(path:String):String=
    loadAsString(this.getClass.getClassLoader().getResource(path))		
  
	
}